public/app/features/logs/utils.ts

```diff
@@ -44,11 +45,18 @@ export function getLogLevel(line: string): LogLevel {
   return level;
 }
 
-export function getLogLevelFromKey(key: string | number): LogLevel {
```

Comment:
@grafana/observability-logs I checked the places where we use this function and it's never called with a number value. Do you know if there's any possibility of receiving a number here?

Reply:
https://github.com/grafana/grafana/blob/b383cafd355172272a30943dd7554646498f8035/public/app/features/logs/logsModel.ts#L446-L451

DataFrame values can basically be `any`, but we should at least plan with number and string, I think. Also, would you mind fixing the conditional in https://github.com/grafana/grafana/blob/b383cafd355172272a30943dd7554646498f8035/public/app/features/logs/logsModel.ts#L447? Now, with `0` being a valid loglevel, this would fail.

Reply:
Perfect, thank you!

Reply:
Great catch on the `if (logLevelKey)`. Should be fixed.